Add raise error for word_tokenize

wannaphong opened the pull request:
Expect to raise error behavior that the input engine does not exist. from #366

wannaphong committed:
Update __init__.py

wannaphong committed:
Change AttributeError to ValueError

from https://github.com/PyThaiNLP/pythainlp/issues/366#issuecomment-592601024

wannaphong committed:
Update test_tokenize.py

pep8speaks commented:
Hello @wannaphong! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2020-02-28 18:37:09 UTC

p16i commented on pythainlp/tokenize/__init__.py:
```diff
@@ -139,10 +139,8 @@ def word_tokenize(
-
-        segments = segment(text, custom_dict)
+    else:
+        raise ValueError()
```
It would be good if we provide some message here. For example, 
```
f"\"{engine}\" not found.  It might be a typo; if not, please consult our document."
```

p16i commented on tests/test_tokenize.py:
```diff
@@ -61,8 +61,8 @@ def test_word_tokenize(self):
-            word_tokenize("หมอนทองตากลมหูว์MBK39", engine="XX")
+        self.assertRaises(
+            ValueError, lambda: word_tokenize("หมอนทองตากลมหูว์MBK39", engine="XX")
         )  # XX engine is not existed
```
Can you also fix the comment? Grammatically, it should be
```
XX engine does not exist.
```

wannaphong committed:
Fix message for ValueError

p16i commented:
LGTM.

wannaphong committed:
Add some test normalize

from https://github.com/PyThaiNLP/pythainlp/commit/1eba27bc8f81ebaa5f04d587881de8c0c1d68f49

wannaphong committed:
Fixed pep8

wannaphong committed:
Fixed pep8 again

wannaphong committed:
Fixed pep8 again ...

wannaphong commented:
@heytitle Thank you.

p16i commented on pythainlp/tokenize/__init__.py:
```diff
@@ -139,10 +139,11 @@ def word_tokenize(
-        segments = segment(text, custom_dict)
+    else:
+        raise ValueError(
+            f"""\"{engine}\""" not found.
```
It seems we have two additional `"` after `\"`.

I purpose we can the statement to
```
raise ValueError(
    f"""Tokenizer \"{engine}\" not found.
It might be a typo; if not, please consult our document."""
)

# output:
ValueError: Tokenizer "XXX" not found.
It might be a typo; if not, please consult our document.
```

wannaphong replied:
> 
> 
> It seems we have two additional `"` after `\"`.
> 
> I purpose we can the statement to
> 
> ```
> raise ValueError(
>     f"""Tokenizer \"{engine}\" not found.
> It might be a typo; if not, please consult our document."""
> )
> 
> # output:
> ValueError: Tokenizer "XXX" not found.
> It might be a typo; if not, please consult our document.
> ```

Fixed #368
